### ISSUE TITLE
Lookup - return single document from target collection

### ISSUE THREAD

**Issue by sterd71** [MorphiaOrg/morphia#1632]
Hi,

I want to peform a lookup to another collection in the same database, using the id as the local field & foreign field, but I only want to return the document from the second collection with the oldest creation date.

I store the creation date on the document so that is available.

I've seen examples in the Mongo documentation where they are building more complex lookups such as:

```
$lookup:
    {
        from: "holidays",
        pipeline: [
            { $match: { year: 2018 } },
            { $project: { _id: 0, date: { name: "$name", date: "$date" } } },
            { $replaceRoot: { newRoot: "$date" } }
        ],
        as: "holidays"
    }
```

Is it possible to use this syntax in Morphia to achieve what I want?

Thanks
Steve

Server Version: 4.2.8 Community
Driver Version: 4.0.3
Morphia Version: 1.6

**Comment by evanchooly** [comment]
You're going to want to upgrade to 2.x to get a complete aggregation API.  What's in 1.6 was an experimental approach I started while I worked at MongoDB that was never finished.  It was not well communicated that the API experimental and folks started relying on it.  The API had some issues that couldn't be fixed without breaking existing applications.  So in 2.whatever, I introduced a more complete API (still marked as `experimental` but will likely get promoted to "final" in the next release or two) that can do what you're asking here.  But it *is* 2.x only so you'll need to update.
